pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains a proxy qubit object for spawning multiple instances of 
+a given qubit type for run on a background scheduler.
+"""
+from typing import List, Union, Tuple, Dict
+from contextlib import nullcontext
+import pennylane as qml
+from pennylane import QubitDevice
+from .default_qubit import DefaultQubit
+from .._version import __version__
+
+try:
+    import dask
+    import dask.distributed as dist
+    from dask.distributed import worker_client, performance_report
+
+    from distributed.protocol import (
+        dask_serialize,
+        dask_deserialize,
+    )
+
+    @dask_serialize.register(qml.numpy.tensor)
+    def serialize(tensor: qml.numpy.tensor) -> Tuple[Dict, List[bytes]]:
+        "Defines a serializer for the Dask backend"
+        header, frames = dist.protocol.numpy.serialize_numpy_ndarray(tensor)
+        header["type"] = "qml.numpy.tensor"
+        header["requires_grad"] = tensor.requires_grad
+        frames = [tensor.data]
+        return header, frames
+
+    @dask_deserialize.register(qml.numpy.tensor)
+    def deserialize(header: Dict, frames: List[bytes]) -> qml.numpy.tensor:
+        "Defines a deserializer for the Dask backend"
+        return qml.numpy.tensor(frames[0], requires_grad=header["requires_grad"])
+
+
+except ImportError as e:  # pragma: no cover
+    raise ImportError("task.qubit requires installing dask and dask.distributed") from e
+
+
+class ProxyInstanceCLS(classmethod):
+    """
+    This utility class allows the use of both an instance
+    as well as class method types. For situations where
+    the explicit class method is needed, we can use this as a
+    decorator, and select the appropriate call dynamically.
+    This is an essential support for accessing backend
+    functionality supports with the proxy `task.qubit`.
+
+    The implementation is based on the Python descriptor guide
+    as mentioned https://stackoverflow.com/questions/28237955/same-name-for-classmethod-and-instancemethod/28238047#28238047
+    """
```

Comment:
This is... really cool 😎 

My only comment here is, from having a read of the SO thread, I'm kinda partial to the second solution given by the answer:

![image](https://user-images.githubusercontent.com/2959003/142985361-b7b674cd-b543-451d-8776-7d057c88a50c.png)


I feel like this leads to cleaner code later on (at the expense of more complex code here), since you can separate out the class and instance methods (and avoid `self_or_cls`).

Reply:
Good point. Yep, the second solution is much cleaner in hindsight. I'll modify to match.